go-controller/pkg/cni/helper_linux.go

```diff
@@ -597,8 +613,17 @@ func (*defaultPodRequestInterfaceOps) ConfigureInterface(pr *PodRequest, getter
 			return nil, fmt.Errorf("unexpected configuration, pod request on dpu host. " +
 				"device ID must be provided")
 		}
+
+		var gwLLA string
+		if pr.IsPrimaryUDN {
+			gwLLA, err = getGWLinkLocalAddress(pr.PodNamespace, pr.PodName, pr.nadName, getter)
```

Comment:
this is doing an extra unmarshalling of annotation for all primary UDN Pods? aren't there any scale implications from doing so?
can we restrict this to only VM pods please?

Reply:
We don't need that since PodAnnotation is already at PodInterfaceInfo, I am going to refactor this.